### PR TITLE
Clean up file system (remove mocking)

### PR DESCRIPTION
**Pull request by hunterwerlla**
## Description

- File system mocking and testing of build in functions is unnecessary, remove both
- Update tests to work with real files
- Update utils to work with correct functions from `fs-extra`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/lambda/local/detectLocalTemplates.test.ts [hunk 1]
```diff
@@ -108,42 +65,9 @@ describe('detectLocalTemplates', async () => {
         assert.strictEqual(templates[0].fsPath, normalizePath(workspaceFolderChildPath, 'template.yaml'))
     })
 
+    // TODO delete this test when this check is fixed
     it('Does not recursively descend past the direct children of each workspace folder', async () => {
-        const workspaceFolderPath = normalizePath('my', 'workspace', 'folder')
```
Comment (awschristou) [inline src/test/lambda/local/detectLocalTemplates.test.ts]: I think this test will never fail in its current form.
Looks like this test needs to create the child, grandchild folders, and write the yaml file to the grandchild folder.

File: src/test/shared/filesystem.test.ts [hunk 2]
```diff
@@ -1,62 +0,0 @@
...
-        it(`filesystem.${fxName} is same as promisify(fs.${fxName})`, async () => {
-            const filesystemFunction = getPropAs<Function>(filesystem, fxName) // filesystem[fxName]
-            const fsFunction = getPropAs<CustomPromisify<Function>>(fs, fxName) // fs[fxName]
-            const actualType = typeof filesystemFunction
-            assert(actualType === 'function', `filesystem.${fxName} should be a "function" but is "${actualType}"`)
-            assert.strictEqual(String(filesystemFunction), String(promisify(fsFunction)))
-        })
-    })
-
-    describe('mkdir', async () => {
```
Comment (awschristou) [inline src/test/shared/filesystem.test.ts]: `mkdir` is still a custom implementation. Seems like we'd still want the mkdir related tests.

File: src/shared/filesystem.ts [hunk 3]
```diff
@@ -3,32 +3,30 @@
...
 
-import * as fs from 'fs'
+import * as fs from 'fs-extra'
 import * as _path from 'path'
-import { promisify } from 'util'
 
-// interfaces & types
-export type PathLike = fs.PathLike
-export type MakeDirectoryOptions = fs.MakeDirectoryOptions
+const mkdir = fs.promises.mkdir
```
Comment (bryceitoc9) [inline src/shared/filesystem.ts]: Is this const needed?

File: src/shared/filesystem.ts [hunk 4]
```diff
@@ -46,25 +44,20 @@ export async function mkdir(
     }
 }
 
-async function mkdirRecursive(path: string, options: MakeDirectoryOptions): Promise<void> {
-    const parent = _path.dirname(path)
-    if (parent !== path) {
-        await mkdir(parent, options)
+export { mkdirSafe as mkdir }
```
Comment (bryceitoc9) [inline src/shared/filesystem.ts]: If the above `const mkdir` isn't needed, just export the `mkdir` function

File: src/shared/filesystemUtilities.ts [hunk 5]
```diff
@@ -27,13 +26,6 @@ export async function fileExists(filePath: string): Promise<boolean> {
     return true
 }
 
-export const readDirAsString = async (
-    pathLike: PathLike,
-    options: { encoding: BufferEncoding } = { encoding: DEFAULT_ENCODING }
```
Comment (bryceitoc9) [inline src/shared/filesystemUtilities.ts]: Is there any change that readdir wouldn't encode as utf8? Looks like that's the main reason why this function existed.

Comment (justinmk3) [reply src/shared/filesystemUtilities.ts]: libuv typically converts filesystem results to utf8, even on Windows, so I would be surprised if readdir returns something else.

Comment (hunterwerlla) [reply src/shared/filesystemUtilities.ts]: By default all the filesystem utils use utf8

File: src/test/shared/filesystem.test.ts (hunk 2 above)
Comment (hunterwerlla) [reply src/test/shared/filesystem.test.ts]: see comment above, this is removed here to pass tests and fixed and re-added by the other PR

File: src/shared/filesystemUtilities.ts (hunk 5 above)
Comment (bryceitoc9) [reply src/shared/filesystemUtilities.ts]: If that's the case, remove `DEFAULT_ENCODING` and `readFileAsString`.

File: src/test/lambda/local/detectLocalTemplates.test.ts (hunk 1 above)
Comment (hunterwerlla) [reply src/test/lambda/local/detectLocalTemplates.test.ts]: Fixed

File: src/test/shared/filesystem.test.ts (hunk 2 above)
Comment (hunterwerlla) [reply src/test/shared/filesystem.test.ts]: This is the wrong file, oops, re-adding test

**Pull request merged**